src/components/ColorRampDropdown.tsx

```diff
@@ -1,25 +1,51 @@
-import React, { ReactElement, useContext, useEffect, useMemo, useState } from "react";
-import { Button, Tooltip } from "antd";
 import { RetweetOutlined } from "@ant-design/icons";
+import { Button, Tooltip } from "antd";
+import React, { ReactElement, useContext, useEffect, useMemo, useState } from "react";
+import { Color } from "three";
 
-import styles from "./ColorRampDropdown.module.css";
-import { DEFAULT_COLOR_RAMPS } from "../constants/color_ramps";
+import { ColorRamp, ColorRampType } from "../colorizer";
+import { ColorRampData, DEFAULT_CATEGORICAL_PALETTES, DEFAULT_COLOR_RAMPS } from "../constants";
 import { AppThemeContext } from "./AppStyle";
+import styles from "./ColorRampDropdown.module.css";
 import IconButton from "./IconButton";
 
 type ColorRampSelectorProps = {
-  selected: string;
-  onChange: (colorRampKey: string, reversed: boolean) => void;
+  selectedRamp: string;
+  onChangeRamp: (colorRampKey: string, reversed: boolean) => void;
   colorRamps?: typeof DEFAULT_COLOR_RAMPS;
+
+  useCategoricalPalettes?: boolean;
+  categoricalPalettes?: typeof DEFAULT_CATEGORICAL_PALETTES;
+  numCategories: number;
+  selectedPalette: Color[];
+  onChangePalette: (newPalette: Color[]) => void;
+
   disabled?: boolean;
   reversed?: boolean;
 };
 
 const defaultProps: Partial<ColorRampSelectorProps> = {
   colorRamps: DEFAULT_COLOR_RAMPS,
   disabled: false,
+  useCategoricalPalettes: false,
+  categoricalPalettes: DEFAULT_CATEGORICAL_PALETTES,
 };
 
+/**
+ * Returns whether if the two arrays are equal, where arr1[i] === arr2[i] for all i.
+ */
```

Comment:
whether if? can't tell if this is an error or if I'm bad at parsing sentences

Reply:
Nope that's absolutely a grammar error from me rewriting this method comment. Thanks for catching this! :X